src/js/hooks/useTablePagination.js

```diff
@@ -11,6 +11,21 @@ const useTablePagination = ({
   });
 
   const onPageChange = (page) => {
+    const maxPage = Math.floor(totalCount / pagination.pageSize);
```

Comment:
Does `maxPage` have to be recalculated on each page change?

Reply:
do you suggest using `useCallback` with totalCount and pageSize as a dependency?

Reply:
yeah, we can use it, this can be cached until filters are applied (`totalCount` changes/page size changes)